vault_core/src/vault.rs

```diff
@@ -342,6 +347,23 @@ impl Vault {
     // Minting and burning
     // ------------------------------------------
 
+    /// Calculate the rewards fee. This is used in `update_vault_balance` to mint
+    /// the `fee` amount in VRTs to the `fee_wallet`.
+    pub fn calculate_rewards_fee(&self, new_balance: u64) -> Result<u64, VaultError> {
```

Comment:
look at stake pool calc_epoch_fee_amount, the returned number needs to be in terms of VRT

Reply:
Done

Reply:
```

    pub fn calculate_rewards_fee(&self, new_balance: u64) -> Result<u64, VaultError> {
        let rewards = new_balance.saturating_sub(self.tokens_deposited);

        let vrt_rewards = self.calculate_vrt_mint_amount(rewards)?;

        if vrt_rewards == 0 {
            return Ok(0);
        }

        let fee = vrt_rewards
            .checked_mul(self.reward_fee_bps as u64)
            .ok_or(VaultError::VaultOverflow)?
            .div_ceil(10_000);

        Ok(fee)
    }
```

Reply:
Is this sufficient?